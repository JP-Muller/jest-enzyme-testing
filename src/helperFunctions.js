import React from 'react'

export const getSum = (num1, num2) => {
    return num1 + num2
}

export const getWeatherIcon = weatherDescription => {
    if (weatherDescription === null) {
      return <i className="far fa-sad-tear" title="Unknown" />;
    } else if (weatherDescription.toLowerCase() === "clear sky") {
      return <i className="far fa-sun" title="Clear Skies" />;
    } else if (weatherDescription.toLowerCase() === "few clouds") {
      return <i className="fas fa-cloud-sun" title="Few Clouds" />;
    } else if (weatherDescription.toLowerCase() === "scattered clouds") {
      return <i className="fas fa-cloud" title="Scattered Clouds" />;
    } else if (weatherDescription.toLowerCase() === "broken clouds") {
      return <i className="fas fa-cloud" title="Broken Clouds" />;
    } else if (weatherDescription.toLowerCase() === "overcast clouds") {
      return <i className="fas fa-cloud" title="Overcast Clouds" />;
    } else if (weatherDescription.toLowerCase() === "shower rain") {
      return <i className="fas fa-cloud-showers-heavy" title="Shower Rain" />;
    } else if (weatherDescription.toLowerCase() === "heavy intensity rain") {
      return <i className="fas fa-cloud-showers-heavy" title="High Intensity Rain" />;
    } else if (weatherDescription.toLowerCase() === "rain") {
      return <i className="fas fa-cloud-rain" title="Rain" />;
    } else if (weatherDescription.toLowerCase() === "light rain") {
      return <i className="fas fa-cloud-rain" title="Light Rain" />;
    } else if (weatherDescription.toLowerCase() === "moderate rain") {
      return <i className="fas fa-cloud-rain" title="Moderate Rain" />;
    } else if (weatherDescription.toLowerCase() === "thunderstorm") {
      return <i className="fas fa-poo-storm" title="Thunderstorm" />;
    } else if (weatherDescription.toLowerCase() === "snow") {
      return <i className="far fa-snowflake" title="Snow" />;
    } else if (weatherDescription.toLowerCase() === "light snow") {
      return <i className="far fa-snowflake" title="Light Snow" />;
    } else if (weatherDescription.toLowerCase() === "mist") {
      return <i className="fas fa-smog" title="Misty" />;
    } else if (weatherDescription.toLowerCase() === "haze") {
      return <i className="fas fa-smog" title="Hazy" />;
    } else if (weatherDescription.toLowerCase() === "smoke") {
      return <i className="fas fa-smog" />;
    } else {
      return <i className="far fa-sad-tear" title="Unknown" />;
    }
  };