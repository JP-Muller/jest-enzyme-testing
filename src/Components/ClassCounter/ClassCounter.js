import React, { Component } from 'react'
import axios from 'axios'
import {getPlaceholderData} from '../../helperFunctions'
import '../../Counters.css'


export class ClassCounter extends Component {
    constructor(props){
        super(props)
        this.state = {
            count: null,
            colorData: [],
            name: '',
            footBallSeason: null
        }
    }

    componentDidMount = () => {
        this.setCount()
        // this.getCurrentFootballSeason()
        getPlaceholderData()
        this.getData()
        this.setName('Jake', 'Muller')
    }

    // componentDidUpdate = (prevState) => {
    //     let {count} = this.state
    //     if (count !== prevState.count)
    // }

    mountCheck = () => {
        console.log('The component was mounted!')
    }

    setName = (first, last) => {
        this.setState({
            name: `${first} ${last}`
        })
    }

    getData = () => {
        this.setState({
            colorData: [
                {
                    color: "red",
                    value: "#f00"
                },
                {
                    color: "green",
                    value: "#0f0"
                },
                {
                    color: "blue",
                    value: "#00f"
                },
                {
                    color: "cyan",
                    value: "#0ff"
                },
                {
                    color: "magenta",
                    value: "#f0f"
                },
                {
                    color: "yellow",
                    value: "#ff0"
                },
                {
                    color: "black",
                    value: "#000"
                }
            ]
        })
    }

    getCurrentFootballSeason = () => {
        return axios.get('https://api.sportsdata.io/v3/nfl/scores/json/CurrentSeason?key=e51c3989c4c643ef85084f27c7e1d466')
        .then(res => {
            return res.data
        })
        .catch(err => console.log('Error', err))
    }

    addNums = (num1, num2) => {
        return num1 + num2
    }

    setCount = () => {
        this.setState({
            count: 0
        })
    }

    incrementCount = () => {
        let {count} = this.state
        this.setState({
            count: count + 1
        })
    }

    decrementCount = () => {
        let {count} = this.state
        this.setState({
            count: count - 1
        })
    }
    render() {
        let {count} = this.state
        return (
            <div className='counter-wrapper'>
                <h3>Class Counter</h3>
                <div className='current-count'>
                        <h1>{count}</h1>
                    </div>
                    <div className='button-container'>
                        <button onClick={() => this.decrementCount()}><i className="far fa-minus-square" /></button>
                        <button onClick={() => this.incrementCount()}><i className="far fa-plus-square" /></button>
                    </div>                
            </div>
        )
    }
}

export default ClassCounter
