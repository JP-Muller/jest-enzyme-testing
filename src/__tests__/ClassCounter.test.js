import React from "react";
import { shallow, mount, render } from "enzyme";
import toJSON from "enzyme-to-json";
import ClassCounter from "../Components/ClassCounter/ClassCounter";
import {getPlaceholderData} from '../helperFunctions'

const setUp = () => {
  const wrapper = shallow(<ClassCounter />);
  return wrapper;
};

describe("<ClassCounter />", () => {
  it("Should render without error", () => {
    const wrapper = setUp();
    expect(toJSON(wrapper)).toMatchSnapshot()
  });

  it("Should render the decrement button", () => {
    const wrapper = setUp();
    const decrementButton = wrapper.find(".fa-minus-square");
    expect(decrementButton.exists()).toBeTruthy();
  });

  it("Should render the increment button", () => {
    const wrapper = setUp();
    const decrementButton = wrapper.find(".fa-plus-square");
    expect(decrementButton.exists()).toBeTruthy();
  });

  it("Does not show count amount on initial render", () => {
    const wrapper = render(<ClassCounter />);
    expect(wrapper.find("h1").text()).toEqual("");
  });

  it("Shows proper count value on mount (Lifecycle check)", () => {
    const wrapper = mount(<ClassCounter />);
    expect(wrapper.find("h1").text()).toEqual("0");
    expect(wrapper.state().count).toEqual(0);
  });

  it("Increments count amount correctly", () => {
    const wrapper = mount(<ClassCounter />);
    wrapper.find(".fa-plus-square").simulate("click");
    expect(wrapper.find("h1").text()).toEqual("1");
    expect(wrapper.state().count).toEqual(1);
  });

  it("Decrements count amount correctly", () => {
    const wrapper = mount(<ClassCounter />);
    wrapper.find(".fa-minus-square").simulate("click");
    wrapper.find(".fa-minus-square").simulate("click");
    wrapper.find(".fa-minus-square").simulate("click");
    expect(wrapper.find("h1").text()).toEqual("-3");
  });

  // it('Calls ComponentDidMount Functions', () => {
  //     const wrapper = mount(<ClassCounter />)
  //     wrapper.instance().mountCheck = jest.fn()
  //     // wrapper.update()
  //     // wrapper.instance().setCount = setCount
  //     // wrapper.instance().setCount()
  //     // wrapper.instance().setCount()
  //     // expect(setCount).toBeCalledTimes(1)
  //     // wrapper.instance().mountCheck()
  //     expect(wrapper.instance().mountCheck).toBeCalledTimes(1)
  // })

  it("Executes displayName properly", () => {
    let wrapper = shallow(<ClassCounter />);
    wrapper.instance().displayName = jest.fn();
    wrapper.instance().displayName("Jake", "Muller");
    console.log(wrapper.instance().displayName("Jake", "Muller"));
    expect(wrapper.instance().displayName).toBeCalledWith("Jake", "Muller");
    expect(wrapper.instance().displayName).toHaveReturnedWith(
      console.log("Jake Muller")
    );
  });

  it("getData returns correct data", () => {
    let wrapper = mount(<ClassCounter />);
    wrapper.instance().getData = jest.fn();
    wrapper.instance().getData();
    expect(wrapper.state().colorData).toEqual([
      {
        color: "red",
        value: "#f00"
      },
      {
        color: "green",
        value: "#0f0"
      },
      {
        color: "blue",
        value: "#00f"
      },
      {
        color: "cyan",
        value: "#0ff"
      },
      {
        color: "magenta",
        value: "#f0f"
      },
      {
        color: "yellow",
        value: "#ff0"
      },
      {
        color: "black",
        value: "#000"
      }
    ]);
  });

  it("Gets correct information from NFL API", () => {
    let wrapper = mount(<ClassCounter />);
    wrapper.instance().getCurrentFootballSeason = jest.fn();
    wrapper.instance().getCurrentFootballSeason();
    expect(wrapper.instance().getCurrentFootballSeason).toBeCalled();
    expect(wrapper.state().footBallSeason).toEqual(2019)
    // expect(wrapper.state().count).toEqual(0)
    // console.log(wrapper.state());
    // console.log(wrapper.props());
  });

  it('Adds 2+2 and returns 4', () => {
    let wrapper = mount(<ClassCounter />)
    wrapper.instance().addNums = jest.fn()
    wrapper.instance().addNums(2,2)
    expect(wrapper.instance().addNums).toHaveBeenCalledWith(2,2)
  })
});
