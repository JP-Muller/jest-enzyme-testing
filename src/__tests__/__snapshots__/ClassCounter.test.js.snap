// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ClassCounter /> Should render without error 1`] = `
<div
  className="counter-wrapper"
>
  <h3>
    Class Counter
  </h3>
  <div
    className="current-count"
  >
    <h1>
      0
    </h1>
  </div>
  <div
    className="button-container"
  >
    <button
      onClick={[Function]}
    >
      <i
        className="far fa-minus-square"
      />
    </button>
    <button
      onClick={[Function]}
    >
      <i
        className="far fa-plus-square"
      />
    </button>
  </div>
</div>
`;
